src/content/1.7/testing/web-acceptance-tests/how-to-contribute-and-create-web-acceptance-tests.md

```diff
@@ -0,0 +1,212 @@
+---
+title: How to contribute and create web acceptance tests
+menuTitle: Contributing to the creation of web acceptance tests
+weight: 2
+---
+
+## How to contribute and create web acceptance tests
+
+
+### Architecture
+[Page Object Model](https://martinfowler.com/bliki/PageObject.html) (also called Page Object Pattern) is a way to organize your code in a test framework. It encourages you to separate your test logic from your page manipulation logic.
+
+Our team uses POM as an architecture to organize the code: on one side we have our page logic, and on the other side we have our test logic.
+
+The goal is to write your test code once, and only change the page logic when someone updates the application.
+
+Both are described in the following paragraphs.
+
+### Pages
+
+#### Name
+The name of the class must be simple and linked to the page in the shop. You should be able to find the corresponding page in the application using the folder hierarchy, and the class name.
+
+The name of the file is a little different: the main page name is always `index.js`.
+
+Example:
+ 
+For BO products pages located at `pages/catalog/products/`, we use 2 classes:
+- `index.js` for products listing page (main page)
+- `add.js` for add/edit product page
+ 
+#### Inheritance
+We heavily use inheritance in the pages to make sure all our generic methods are available from everywhere, and you don’t need to instantiate separate objects. 
+
+There is a main mother class, called `CommonPage`, located at the root of the `pages` folder.
+
+We have another level of inheritance in each root folder (`BO` and `FO`) through the classes `BOmainPage.js` and `FOMainPage.js`. Of course, `CommonPage` is **not** meant to be instantiated (it’s kind of abstract class) !
+
+When you create new page classes, make sure to make them inherit from their respective `XXMainPage` to be sure to be able to use both generic methods and specific methods.
+
+Examples: 
+
+- Install page (which is independent) extends from `CommonPage`
+- Brands page (which is part of the BO) extends from `BOBasePage`
+- Checkout page (which is part of the FO) extends from `FOBasePage`  
+
+#### Methods
+Methods inside a page class must be properly named about their primary function. Remember to [KISS](https://en.wikipedia.org/wiki/KISS_principle) ! You should be able to understand what a method does just by looking at its name.
+
+Methods inside a page class **CANNOT USE** the `expect` keyword or any kind of asserting function, since their sole purpose is to interact with the page they’re linked with. They don’t validate anything per se, they just expose functionalities. 
+To validate a behavior in your test logic, your methods can return booleans, integers (number of lines, elements, etc), strings, objects... It’s up to you.
+
+Examples: 
+```
```

Comment:
```suggestion
```js
```

Reply:
This how it display **with** js 
![image](https://user-images.githubusercontent.com/48441421/80367595-aea37100-888b-11ea-9482-5f5a91611c47.png)

And this is how it display **without** js
![image](https://user-images.githubusercontent.com/48441421/80367709-dbf01f00-888b-11ea-8997-c210aac670f5.png)

For better visibility, i choose to not use it, WDYT ?

Reply:
That's quite odd... see, we use `js` already in some pages like [this one](https://github.com/PrestaShop/docs/blob/master/src/content/1.7/development/components/grid/columns-reference/position.md#L46) and it [renders well](https://devdocs.prestashop.com/1.7/development/components/grid/columns-reference/position/):

<img width="763" alt="Screenshot 2020-04-30 at 17 42 41" src="https://user-images.githubusercontent.com/1009343/80730559-051de300-8b0a-11ea-9835-b001c5128ddb.png">

Reply:
I think it's broken when testing in local machine, so I added js